### PR TITLE
Make Map::map_can_construct_*() use CoordsXYRangedZ

### PR DESCRIPTION
**Pull request by tupaschoal**

### CODE REVIEW

File: src/openrct2/peep/Peep.cpp [hunk 1]
```diff
@@ -781,7 +781,7 @@ bool Peep::Place(TileCoordsXYZ location, bool apply)
         return false;
     }
 
-    if (!map_can_construct_at(destination.x, destination.y, destination.z / 8, (destination.z / 8) + 1, { 0b1111, 0 }))
+    if (!map_can_construct_at({ destination, destination.z, destination.z + 1 * 8 }, { 0b1111, 0 }))
```
Comment (duncanspumpkin) [inline src/openrct2/peep/Peep.cpp]: brackets around the `(1 * 8)`

File: src/openrct2/world/Map.cpp [hunk 2]
```diff
@@ -1369,38 +1370,38 @@ bool map_can_construct_with_clear_at(
...
-                bh = zLow + 4;
+                bh = pos.baseZ / 8 + 4;
                 {
                     auto baseQuarter = quarterTile.GetBaseQuarterOccupied();
                     auto zQuarter = quarterTile.GetZQuarterOccupied();
-                    if ((!(baseQuarter & 0b0001) || ((zQuarter & 0b0001 || zLow >= al) && bh >= al))
-                        && (!(baseQuarter & 0b0010) || ((zQuarter & 0b0010 || zLow >= ah) && bh >= ah))
-                        && (!(baseQuarter & 0b0100) || ((zQuarter & 0b0100 || zLow >= cl) && bh >= cl))
-                        && (!(baseQuarter & 0b1000) || ((zQuarter & 0b1000 || zLow >= ch) && bh >= ch)))
+                    if ((!(baseQuarter & 0b0001) || ((zQuarter & 0b0001 || pos.baseZ / 8 >= al) && bh >= al))
```
Comment (duncanspumpkin) [inline src/openrct2/world/Map.cpp]: For future refactoring job `al`, `ah`, `cl`, `ch` should be renamed `northZ`, `eastZ`, `southZ`, westZ` and be moved to bigZ.

Comment (tupaschoal) [reply src/openrct2/world/Map.cpp]: I've noted to do it later.

**Comment by tupaschoal** [comment]
Dealt with issues.

**Pull request merged**